File: THCS/2022_2023/Hue_2223/Bai2/DAY.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <iomanip> // Để sử dụng std::setprecision

int main() {
    // Đọc dữ liệu từ file DAY.INP
    std::ifstream inputFile("DAY.INP");
    int n;
    inputFile >> n; // Số công nhân

    std::vector<int> daysWorked(n);
    for (int i = 0; i < n; ++i) {
        inputFile >> daysWorked[i]; // Số ngày làm việc của mỗi công nhân
    }
    inputFile.close();

    // Sắp xếp số ngày làm việc theo thứ tự giảm dần
    std::sort(daysWorked.rbegin(), daysWorked.rend());

    // Tính tổng số ngày làm việc
    int totalDays = 0;
    for (int days : daysWorked) {
        totalDays += days;
    }

    // Tính số ngày làm việc trung bình
    double averageDays = static_cast<double>(totalDays) / n;

    // Ghi kết quả vào file DAY.OUT
    std::ofstream outputFile("DAY.OUT");
    for (int i = 0; i < n; ++i) {
        outputFile << daysWorked[i] << (i < n - 1 ? " " : "");
    }
    outputFile << std::endl;
    outputFile << "Tong: " << totalDays << "; trung bình: " << std::fixed << std::setprecision(2) << averageDays;
    outputFile.close();

    return 0;
}

File: THCS/2022_2023/Hue_2223/Bai2/DAY.py
# Đọc dữ liệu từ file DAY.INP
with open('DAY.INP', 'r') as file:
    n = int(file.readline().strip())  # Số công nhân
    days_worked = list(map(int, file.readline().strip().split()))  # Số ngày làm việc của mỗi công nhân

# Sắp xếp số ngày làm việc theo thứ tự giảm dần
days_worked.sort(reverse=True)

# Tính tổng số ngày làm việc
total_days = sum(days_worked)

# Tính số ngày làm việc trung bình
average_days = total_days / n

# Ghi kết quả vào file DAY.OUT
with open('DAY.OUT', 'w') as file:
    # Ghi số ngày làm việc đã sắp xếp
    file.write(' '.join(map(str, days_worked)) + '\n')
    # Ghi tổng số ngày làm việc và số ngày làm việc trung bình
    file.write(f'Tong: {total_days}; trung bình: {average_days:.2f}')
